### PR TITLE
Use Self where possible

### PR DESCRIPTION
**Pull request by adamnemecek**
Retry of #433

### CODE REVIEW

File: vendored/sqlite3-parser/src/lexer/sql/mod.rs [hunk 1]
```diff
@@ -38,7 +38,7 @@ impl<'input> Parser<'input> {
         let scanner = Scanner::new(lexer);
         let ctx = Context::new(input);
         let parser = yyParser::new(ctx);
-        Parser {
+        Self {
```
Comment (jussisaurio) [inline vendored/sqlite3-parser/src/lexer/sql/mod.rs:41]: I think this PR looks good but can you remove the modifications to `vendored`?

**Comment by adamnemecek** [comment]
Ok fixed and merged your changes.

File: vendored/sqlite3-parser/src/parser/mod.rs [hunk 2]
```diff
@@ -68,7 +68,7 @@ pub struct Context<'input> {
 
 impl<'input> Context<'input> {
     pub fn new(input: &'input [u8]) -> Self {
-        Context {
+        Self {
```
Comment (jussisaurio) [inline vendored/sqlite3-parser/src/parser/mod.rs:71]: theres still a change to vendored

**Comment by adamnemecek** [comment]
Why was this closed?

**Comment by jussisaurio** [comment]
it was merged, pekka's merge-pr.py script sometimes ends up not showing it as merged in github, but you can see the commit hash above https://github.com/tursodatabase/limbo/commit/97647ff0568f2a160bf3f639440a592a18881fab

**Comment by adamnemecek** [comment]
Ah ok.

**Comment by penberg** [comment]
Actually, this one I edited manually to drop the vendored changes and merged by hand because it was about to bit-rot again as we merge other PRs.

**Pull request closed without merging**